### PR TITLE
Fixed permission checks when managing/deleting a thread

### PR DESCRIPTION
**Pull request by Almighty-Satan**
[contributing]: https://github.com/DV8FromTheWorld/JDA/wiki/5%29-Contributing

## Pull Request Etiquette

- [X] I have checked the PRs for upcoming features/bug fixes.
- [X] I have read the [contributing guidelines][contributing].

### Changes

- [X] Internal code
- [ ] Library interface (affecting end-user code) 
- [X] Documentation
- [ ] Other: \_____ 

Closes Issue: #2188 

## Description

When calling `ThreadChannel#getManager` or `ThreadChannel#delete` it is currently checked if the user has the `MANAGE_CHANNEL` permission, but the permission actually required to manage or delete a thread is `MANAGE_THREADS`.

### CODE REVIEW

File: src/main/java/net/dv8tion/jda/api/entities/GuildChannel.java [hunk 1]
```diff
@@ -35,12 +36,23 @@ public interface GuildChannel extends Channel, Comparable<GuildChannel>
...
     ChannelManager<?, ?> getManager();
+    
+    /**
+     * Returns the {@link Permission} needed in order to manage or delete this channel.
+     * This returns {@link Permission#MANAGE_CHANNEL} or {@link Permission#MANAGE_THREADS} for {@link ThreadChannel ThreadChannels}
+     * 
+     * @return The {@link Permission}
+     */
+    @Nonnull
+    default Permission getManagePermission() {
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/api/entities/GuildChannel.java:53]: I'm against this design, I think instead we should pass the required permissions to the manager using `EnumSet<Permission>` as a constructor parameter.

Also, please try to follow the style of the project.

Comment (Almighty-Satan) [reply src/main/java/net/dv8tion/jda/api/entities/GuildChannel.java:53]: Why use a `Set`? Do you think Discord will require multiple permissions for editing channels in the future?

**Comment by DV8FromTheWorld** [comment]
Bump. This feature seems simple, can we get it across the finish line?

**Comment by DV8FromTheWorld** [comment]
TLDR: We aren't going to go with this approach and instead update the manager check only.

An enumset here would provide more future proofing than a single Permission return would. While right now only a single Permission is required, this could change in the future, and since permissions are represented as a Set everywhere else, thatd make the most sense here as well.

However, in reality, this check is now more complicated than just a single Permission check. For threads, we also need to be aware of who owns the thread when determining if it is manageable. You don't need MANAGE_THREADS if you own the thread.

As such, we are going to entirely scrap the getter as it cannot represent that complex logic via a single Permission return, nor an EnumSet return.
